File: app/src/main/java/com/kakeibo/data/disk/Converters.java
package com.kakeibo.data.disk;

import androidx.room.TypeConverter;
import com.kakeibo.util.UtilCurrency;
import java.math.BigDecimal;

public class Converters {
    @TypeConverter
    public long bigDecimalToLong(BigDecimal input) {
        return UtilCurrency.getLongFromBigDecimal(input.multiply(new BigDecimal(1000)));
    }

    @TypeConverter
    public BigDecimal longToBigDecimal(long input) {
        return BigDecimal.valueOf(input);
    }
}

File: app/src/main/java/com/kakeibo/data/disk/Converters.kt
package com.kakeibo.data.disk

import androidx.room.TypeConverter
import com.kakeibo.R
import com.kakeibo.SubApp
import java.math.BigDecimal

class Converters {
    @TypeConverter
    fun bigDecimalToLong(input: BigDecimal): Long {
        return input.multiply(BigDecimal(1000)).toLong()
    }

    @TypeConverter
    fun longToBigDecimal(input: Long): BigDecimal {
        return BigDecimal.valueOf(input)
                .divide(BigDecimal.valueOf(1000),
                        SubApp.getFractionDigits(R.string.pref_key_fraction_digits),
                        BigDecimal.ROUND_HALF_UP)
    }
}